Koha/Illrequest.pm

```diff
@@ -1346,57 +1330,282 @@ EOF
             "No target email addresses found. Either select at least one partner or check your ILL partner library records.")
           if ( !$to );
         # Create the from, replyto and sender headers
-        my $from = $branch->branchemail;
+        my $from = $branch->branchillemail || $branch->branchemail;
```

Comment:
We should revert this.. 'from' should always be a valid 'from' address that the koha server has permission to send mail from.